ledger/ledger.go

```diff
@@ -4,15 +4,12 @@ import (
 	"context"
 	"time"
 
-	ic "github.com/libp2p/go-libp2p-core/crypto"
 	ledgerpb "github.com/tron-us/go-btfs-common/protos/ledger"
+	"github.com/tron-us/go-btfs-common/utils/grpc"
 	"github.com/tron-us/protobuf/proto"
-	"google.golang.org/grpc"
-)
 
-func NewClient(conn *grpc.ClientConn) ledgerpb.ChannelsClient {
```

Comment:
I still prefer having a `NewClient` func, but instead of accepting a new conn, you can do `client := NewClient(addr string)`

Then you can do `client.ImportAccount(ctx, pubKey)`, by doing this, you actually complete BTFS-1001.

Reply:
actually, we have a `grpc.LedgerClient(addr).WithContext(...)`.

Reply:
ok, but `addr` needs to be passed into every single helper here. If that's what Escrow feels most comfortable .. sure.

Reply:
> ok, but `addr` needs to be passed into every single helper here. If that's what Escrow feels most comfortable .. sure.
since we uses short connection, we have to do so. i've done the escrow side refactoring, you can take a look at: `https://github.com/TRON-US/go-btfs-escrow/pull/17/files#diff-0a50d3fec2f1723d3f3e08e6ae2ed64cR20`

Reply:
Yes, addr has to be passed in. I was suggesting an alternative setup (same logic), where `ImportAccount` does not need to have `addr` passed in as an argument, because arguments should be non-connection related business logic. By having `func (client *SomeClient) ImportAccount(ctx context.Context, pubKey ic.PubKey)` it's cleaner way of binding configurations to an object, and define a call on that. Still will pass `client.addr` to the grpc connection, but won't need to pass `addr` to the `ImportAccount` function. This is a similar setup to the client builder way.

Reply:
oh, i see, it's a good idea, will do.